src/Symfony/Component/Validator/Tests/Constraints/UrlValidatorTest.php

```diff
@@ -74,6 +74,8 @@ public function getValidUrls()
             ['http://www.google.museum'],
             ['https://google.com/'],
             ['https://google.com:80/'],
+            ['http://goog_le.com'],
```

Comment:
Is it possible to have an underscore in the domain name itself?

Reply:
it depends if we check for domain name vs. hostname ([ref](https://stackoverflow.com/questions/2180465/can-domain-name-subdomains-have-an-underscore-in-it#answer-2183140)), so this is also related to #31518 which assumed a hostname in fact :thinking:

Reply:
Indeed. A simple example of this is if you take a known IP address (like 72.30.35.10 which belongs to Yahoo) and add that to your hosts file against a hostname with an underscore. 

````
72.30.35.10         goog_le.com
````

Which loads as a valid URL:

![Screen Shot 2019-07-15 at 9 15 43 PM](https://user-images.githubusercontent.com/2110222/61219236-607f1000-a746-11e9-8488-2ad2f81bc1d5.png)

Reply:
The issue is about an underscore in a domain name, not the hostname.

While patching `/etc/host/ may work, it can also lead to side effects, e.g. https://kb.vmware.com/s/article/2048670

From https://github.com/symfony/symfony/issues/32506#issuecomment-510899729

> you can browse the website ... so "it doesn't matter" if some RFC says this is not valid

In this case we created the domain name in /etc/hosts, im not sure it still qualifies as "valid", nor if the intend of the URL validator is to allow any "url that some client may load".

I tend to prefer the RFC validation, yet the issue about underscores in a subdomain is still valid.